File: app/src/test/java/com/example/java_unittest_mitch/util/DateUtilTest.java
package com.example.java_unittest_mitch.util;

import static com.example.java_unittest_mitch.util.DateUtil.monthNumbers;
import static com.example.java_unittest_mitch.util.DateUtil.months;
import static org.junit.jupiter.api.Assertions.*;

import androidx.core.widget.TintableImageSourceView;

import com.example.java_unittest_mitch.models.Note;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;
import org.junit.jupiter.api.TestReporter;
import org.junit.jupiter.api.function.Executable;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import java.text.SimpleDateFormat;
import java.util.Date;


class DateUtilTest {
    private static final String today="04-2022";

    @Test
    public void getCurrentTime() {
        Assertions.assertDoesNotThrow(new Executable() {
            @Override
            public void execute() throws Throwable {
                assertEquals(today,DateUtil.getCurrentTimeStamp());
            }
        });
    }
    //way simple but not enough for testing a parameters of a list
    @Test
    public void test_getMonthFromNumber() {
        assertEquals(months[3],DateUtil.getMonthFromNumber("04"));
    }

    //best way,parameterized way
    @ParameterizedTest
    @ValueSource(ints = {0,1,2,3,4,5,6,7,8,9,10,11})
    public void test_monthFromNumberForAllPossibleValues(int monthNumber
            , TestInfo testInfo, TestReporter testReporter){
        assertEquals(months[monthNumber],DateUtil.getMonthFromNumber(monthNumbers[monthNumber]));
    }




}

File: app/src/test/java/com/example/java_unittest_mitch/util/DateUtilTest.kt
package com.codingwithmitch.unittesting2.util

import com.example.java_unittest_mitch.util.DateUtil.*
import org.junit.jupiter.api.*

import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource
import kotlin.random.Random

class DateUtilTest
{
    private val today="01-2023"
    /*to be sure that exception of date-format is not going to be created*/
    @Test
    internal fun notThrowException_when_dateFormat_isCorrect()
    {
        assertDoesNotThrow()
        {
            Assertions.assertEquals(today,  getCurrentTimeStamp())
        }
    }

    //-------------------------------------------------------------------

    @ParameterizedTest
    @ValueSource(ints = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11])
    internal fun monthNumbersAreCorrect(monthNumber:Int,
                    testInfo: TestInfo,testReporter:TestReporter)
    {
        Assertions.assertEquals(months[monthNumber], getMonthFromNumber(monthNumbers[monthNumber]))

    }
    //----------------------------------------------------------------------
    @ParameterizedTest
    @ValueSource(ints = [1,2,3,4,5,6,7,8,9,10,11,12])
    internal fun returnError_when_positions_isNotCorrect(number:Int)
    {
        val randomNumber= Random.nextInt(90)+12
        Assertions.assertEquals(getMonthFromNumber(randomNumber.toString()),GET_MONTH_ERROR)

    }



}